pkg/virt-api/rest/subresource.go

```diff
@@ -288,6 +281,18 @@ func (app *SubresourceAPIApp) StartVMRequestHandler(request *restful.Request, re
 		return
 	}
 
+	vmi, err := app.VirtCli.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			response.WriteError(http.StatusInternalServerError, err)
+			return
+		}
+	}
+	if vmi != nil && vmi.IsRunning() {
```

Comment:
Oops, I should have noticed this earlier. in this context "IsRunning" is only true if the VMI is in phase==Running, so this is probably too narrow a check. This check should probably be extended to apply to any non-final state... Anything in "pending, scheduling or scheduled" will likely soon be "running". How about something like this?

`if vmi != nil && !vmi.IsFinal() && vmi.Phase != v1.Unknown {`